### PR TITLE
models: adding support for gemini and api keys

### PR DESCRIPTION
**Pull request by ishandhanani**
PR adds support for the google LLM family and allows API keys to be passed through the config.yaml file

### CODE REVIEW

File: config.go [hunk 1]
```diff
@@ -15,11 +15,13 @@ var defaultModels = map[string]string{
 	"anthropic": "claude-3-5-sonnet-20240620",
 	"openai":    "gpt-4o",
 	"ollama":    "llama3",
+	"googleai":  "gemini-pro",
 }
 
 // Config is the configuration for cgpt.
 type Config struct {
 	Backend   string `yaml:"backend"`
+	APIKey    string `yaml:"apiKey"`
```
Comment (tmc) [inline config.go:24]: Should indicate vendor-specificity, I think.

**Review by tmc — APPROVED** [review]

File: config.go (hunk 1 above)
Comment (ishandhanani) [reply config.go]: Added separate api key fields for each provider

File: config.go [hunk 2]
```diff
@@ -15,14 +15,18 @@ var defaultModels = map[string]string{
...
 }
 
 // Config is the configuration for cgpt.
 type Config struct {
-	Backend   string `yaml:"backend"`
-	Model     string `yaml:"modelName"`
-	Stream    bool   `yaml:"stream"`
-	MaxTokens int    `yaml:"maxTokens"`
+	Backend           string `yaml:"backend"`
+	OPENAI_API_KEY    string `yaml:"OPENAI_API_KEY"`
```
Comment (tmc) [inline config.go:24]: kind of odd to capitalize these

**Review by tmc — APPROVED** [review]
thanks, I'm going to change this up a bit but accepting.

**Pull request merged**